CustomShips.cpp

```diff
@@ -472,6 +472,27 @@ HOOK_METHOD(ShipManager, OnLoop, () -> void)
             ex->erosion.anim->OnUpdate();
         }
     }
+    
+    bool hackSoundLoop = false;
+    if (hackingSystem)
+    {
+        hackSoundLoop = hackingSystem->SoundLoop();
+    }
+    if (!hackSoundLoop)
+    {
+        ShipManager* otherShip = G_->GetShipManager((iShipId + 1)%2);
```

Comment:
iShipID + 1 modulo 2?

Why not just `G_->GetShipManager(1)` since that's the enemy ship.
`0` is the player ship, probably ideal if we had an enumeration for iShipId but it's whatever since it commonly comes from the game.

------

Or I see you're just trying to get the other one, then in that case *yes the modulo works* but this makes for less readable code since it's not needed, a simple ternary might be nicer for readability like:

`iShipId == PLAYER ? ENEMY : PLAYER` (assuming we had an enumeration for it, which I don't think we do)
so
`iShipId == 1 ? 0 : 1` otherwise but at least it's clear we're only selecting one or the other ship?


I dunno, it just *feels* wrong to me like the modulo is because of the lack of an enumeration and if there was an enumeration the module would be nuts to do, so I feel it would be cleaner to treat it as if it was and maybe we can clean that up later to use an enumeration (or do that now) so it's clearer to everyone reading it that we're just selecting one of two ships.

Reply:
Fixed